### ISSUE TITLE
Media replace file mqtt publish by mod_mqtt is not allowed for anonymous users

### ISSUE THREAD

**Issue by robvandenbogaard** [zotonic/zotonic#1988]
This is a problem in cases where media files (viewable by anonymous users) are being converted/rendered on-demand, and the view needs to get updated when they are available for display.

**Comment by mworrell** [comment]
@robvandenbogaard could you copy the code we made for this?  Then I'll add it to the core.

**Comment by robvandenbogaard** [comment]
```
observe_acl_is_allowed(#acl_is_allowed{
        action = subscribe,
        object = #acl_mqtt{ words=[<<"site">>, _, <<"rsc">>, RscId, <<"medium">>] }
    }, Context) ->
    z_acl:rsc_visible(z_convert:to_integer(RscId), Context);
observe_acl_is_allowed(_, _Context) ->
    undefined.
```

**Comment by mworrell** [comment]
In the 0.x this is part of the default ACL lookups:

```
is_allowed(subscribe, _Topic, [<<"site">>, Site, <<"rsc">>, RscId | _], Site, Context) ->
    z_acl:rsc_visible(z_convert:to_integer(RscId), Context);
```

Could you check if the 0.x works well for your use case?

**Comment by mworrell** [comment]
@robvandenbogaard kan je dit checken?

**Comment by mworrell** [comment]
I get the expected result:
```
(z0x@PoToi)4> z_mqtt_acl:is_allowed(subscribe, <<"~site/rsc/1/medium">>, z:c(test0x)).
true
```

Closing this issue, as it is fixed in the 0.46.
